File: src/main/java/me/miki/shindo/injection/mixin/interfaces/client/gui/IMixinChatLine.java
package me.miki.shindo.injection.mixin.interfaces.client.gui;

import java.lang.ref.WeakReference;
import java.util.HashSet;

public interface IMixinChatLine {
    HashSet<WeakReference<Object>> chatLines = new HashSet<>();

    Object client$getPlayerInfo();
}

File: src/main/kotlin/me/miki/shindo/injection/mixin/interfaces/client/gui/IMixinChatLine.kt
package me.miki.shindo.injection.mixin.interfaces.client.gui

import kotlin.jvm.JvmField
import java.lang.ref.WeakReference
import java.util.HashSet
import net.minecraft.client.gui.ChatLine
import net.minecraft.client.network.NetworkPlayerInfo

interface IMixinChatLine {
    companion object {
        @JvmField
        val chatLines = HashSet<WeakReference<ChatLine>>()
    }

    fun `client$getPlayerInfo`(): NetworkPlayerInfo
}
